serialization.h

```diff
@@ -242,3 +244,76 @@ inline void from_json(const nlohmann::json& j, unique_ptr<ISocketChannel>& socke
     );
 }
 
+inline void to_json(nlohmann::json &j, const IController &controller)
+{
+    try
+    {
+        j["port"] = controller.GetPort();
+        j["canvases"] = nlohmann::json::array();
+        
+        for (const auto ptrCanvas : controller.Canvases())
+        {
+            nlohmann::json canvasJson;
+            to_json(canvasJson, ptrCanvas);           // Uses the Canvas serializer
+            j["canvases"].push_back(canvasJson);
+        }
+    }
+    catch (const std::exception &e)
+    {
+        j = nullptr;
+    }
+}
+
+// I've run into the same dependency loop issue again here... I'm not sure how to resolve it!
+// I think I need to move the from_json function to a separate file, but I know you want to keep
+// it here likely.  It can go right into Controller.h nicely though....
```

Comment:
No, I actually don't want to keep it here. I'd like us to make a decision on where to keep (de)serialization logic for types we declare ourselves, and stick with it.

Personally, I think we should keep the (de)serialization logic for a class/struct as close to the type's declaration as possible. This is also consistent with the approach we used in NDS, where the logic is actually _embedded_ in the classes/structs in question. 

In that scenario, serialization.h would only contain (de)serialization logic for a few (basic) types that are declared externally - again the same way we do things in NDS.

I'm happy to push an update to the Thursday branch to implement this refactor throughout the project, if you're okay with that. It would mean I remove the interface.h and serialization.h includes from monitor/main.cpp and monitor/content.cpp; we already concluded including those headers there is moot anyway.

Reply:
Yes, please do.  What you're describing is what I'd originally planned, but then tried to make it a priority to keep it all together, which didn't work out!

Reply:
Done. PR #16 aims to modify this PR to do exactly that. Which means that we can proceed with this PR when that one is merged.